solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/GabrielaSillis/GSLoginCoordinator.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  GSCoordinator.swift
+//  DeliveryAppChallenge
+//
+//  Created by Gabriela Sillis on 20/07/22.
+//
+
+import UIKit
+
+protocol GSLoginCoordinating {
+    func startNavigatingFlow()
+    func displayErrorAlertView()
+    func navigatingToResetPassword()
+    func navigatingToCreateAccount()
+    var controller: UIViewController? { get set }
+}
+
+struct GSLoginCoordinator: GSLoginCoordinating {
+    var controller: UIViewController?
+    
+    func startNavigatingFlow() {
+        let viewController = UINavigationController(rootViewController: HomeViewController())
+        let scenes = UIApplication.shared.connectedScenes
+        let windowScene = scenes.first as? UIWindowScene
+        let window = windowScene?.windows.first
+        window?.rootViewController = viewController
+        window?.makeKeyAndVisible()
+    }
+    
+    func displayErrorAlertView() {
+        let alertController = UIAlertController(
+            title: "Ops..",
+            message: "Houve um problema, tente novamente mais tarde.",
+            preferredStyle: .alert
+        )
+        let action = UIAlertAction(
+            title: "Ok",
+            style: .default
+        )
+        alertController.addAction(action)
+        controller?.present(alertController, animated: true)
+    }
+    
+    func navigatingToResetPassword() {
+        let storyboard = UIStoryboard(name: "GSUser", bundle: nil)
+        let vc = storyboard.instantiateViewController(withIdentifier: "GSResetPasswordViewController") as! GSResetPasswordViewController
```

Comment:
uma dica: sempre que for trabalhar com um codigo e tiver um force unwrap, já refatore pra n usa-lo
sugeriria um guard let aqui

Reply:
nesse caso, se nao achar o storyboard eu coloquei um fatalError com uma mensagem no tratamento do guard let, acha que essa é uma abordagem adequada?

Reply:
o ideal seria n usar storyboard,
como essa n é task de remover storyboard, o ideal seria mostrar um alerta padrão pro usuario ("algo de errado aconteceu")
o fatalError erro no geral não é uma opção, pq crasha o aplicativo, e isso deve ser sempre evitado para o usuario